### PR TITLE
Splitter class sprint 1

### PR DESCRIPTION
**Pull request by dgh007786**
This is a simple working code for split by percentage, random state, shuffle, and stratify.

### CODE REVIEW

File: simple_classifier/Splitter/Splitter.py [hunk 1]
```diff
@@ -2,7 +2,39 @@
...
+    @abc.abstractmethod
     def split_data(self, X, y):
-        raise NotImplementedError(
-            "Subclasses must implement split_data method")
+        """
+        Input:
+        ------
+
+        X : array
+        X is the feature matrix. It is typincally two dimentional array or matrix of shape (n_samples,n_features)
```
Comment (briancatraguna) [inline simple_classifier/Splitter/Splitter.py:29]: typo `typically`

File: simple_classifier/Splitter/Splitter_subclasses.py [hunk 2]
```diff
@@ -0,0 +1,42 @@
+from simple_classifier.Splitter.Splitter import Splitter
+from sklearn.model_selection import train_test_split
+
+
+@Splitter.register("Percentage_split")
```
Comment (briancatraguna) [inline simple_classifier/Splitter/Splitter_subclasses.py:5]: Let's keep codenames all in lower case for consistency -> `percentage_split`. What do you think?

File: simple_classifier/Splitter/Splitter_subclasses.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+    """
+    def split_data(self, X, y):
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=self.test_size,
+                                                            random_state=self.random_state, shuffle=False,
+                                                            stratify=None)
+        return X_train, X_test, y_train, y_test
+
+
+@Splitter.register("Percentage_split_w_shuffle")
```
Comment (briancatraguna) [inline simple_classifier/Splitter/Splitter_subclasses.py:19]: `percentage_split_w_shuffle`

File: simple_classifier/Splitter/Splitter_subclasses.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+class PercentageShuffleSplitter(Splitter):
+
+    def split_data(self, X, y):
+        X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=self.test_size,
+                                                            random_state=self.random_state, shuffle=True, stratify=None)
+        return X_train, X_test, y_train, y_test
+
+
+@Splitter.register("Percentage_split_w_stratify")
```
Comment (briancatraguna) [inline simple_classifier/Splitter/Splitter_subclasses.py:28]: `percentage_split_w_stratify`

**Review by briancatraguna — CHANGES_REQUESTED** [review]
Overall looks great man, great job. Just added some minor comments.

Also as a group, I think we need to decide on the docstring convention. There are different docstring conventions we can choose, each has different formatting and wordings:
Screen Shot 2023-05-05 at 19 54 31

We need to keep it consistent through out the whole codebase so that when it is generated by sphinx it'll be easier to follow and read as its consistent. 

I'm not sure if sphinx also has their own docstring convention. @Alan052918 can you help confirm?

**Review by jayeshgaur — APPROVED** [review]

**Comment by Alan052918** [comment]
Sphinx docstring format: https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

**Review by Alan052918 — CHANGES_REQUESTED** [review]
- Need another splitter (or a parameter to all splitters in this file) that take a list of indices as seed from the user and further split the rest of the data based on this seed.
- Nitpick: Could extract each `Splitter` subclass into its own file. Idiomatically python file names are all lowercases delimited by underscores, but since Brian named files in PascalCase in the skeleton we can stick to it. But you could rename this file as `SplitterSubclasses` to keep consistency.

**Review by briancatraguna — APPROVED** [review]

**Review by Alan052918 — APPROVED** [review]
